torch/distributed/tensor/_random.py

```diff
@@ -38,7 +40,7 @@ def is_rng_supported_mesh(device_mesh: DeviceMesh) -> bool:
         Currently we only support correct RNG on cuda/cuda-like devices.
     """
     device_handle = _get_device_handle(device_mesh.device_type)
-    if device_handle and hasattr(device_handle, "set_rng_state"):
+    if device_handle and hasattr(device_handle, "rng_supported_mesh"):
```

Comment:
Here only check whether the device has this attribute, but does not care about the value of this attribute.
Does this design confuse PyTorch user and developer?

Reply:
Modified to return actual attribute value